storm-core/src/jvm/org/apache/storm/topology/CheckpointTupleForwarder.java

```diff
@@ -53,6 +53,10 @@
     private long lastTxid = Long.MIN_VALUE;
     private AnchoringOutputCollector collector;
 
+    public CheckpointTupleForwarder() {
```

Comment:
Do we need this constructor? Especially it will incur NPE since this.bolt is accessed from various methods.